cli/src/main/java/org/projectnessie/tools/catalog/migration/cli/CatalogMigrationCLI.java

```diff
@@ -0,0 +1,507 @@
+                  + "Please check the `catalog_migration.log` file for the reason. "
+                  + "%nFailed to delete identifiers are written into `%s`. ",
+              result.failedToDeleteTableIdentifiers().size(),
+              sourceCatalogType,
+              FAILED_TO_DELETE_AT_SOURCE_FILE));
+    }
+  }
+
+  private void printDetails(CatalogMigrationResult result, PrintWriter printWriter) {
+    printWriter.println(String.format("%nDetails: "));
+    if (!result.registeredTableIdentifiers().isEmpty()) {
+      printWriter.println(
+          String.format(
+              "- Successfully %s these tables:",
+              deleteSourceCatalogTables ? "migrated" : "registered"));
+      printWriter.println(result.registeredTableIdentifiers());
+    }
+
+    if (!result.failedToRegisterTableIdentifiers().isEmpty()) {
+      printWriter.println(
+          String.format(
+              "- Failed to %s these tables:", deleteSourceCatalogTables ? "migrate" : "register"));
+      printWriter.println(result.failedToRegisterTableIdentifiers());
+    }
+
+    if (!result.failedToDeleteTableIdentifiers().isEmpty()) {
+      printWriter.println("- [WARNING] Failed to delete these tables from source catalog:");
+      printWriter.println(result.failedToDeleteTableIdentifiers());
+    }
+  }
+
+  private void printDryRunResults(CatalogMigrationResult result, PrintWriter printWriter) {
+    printWriter.println(String.format("%nSummary: "));
+    if (result.registeredTableIdentifiers().isEmpty()) {
+      printWriter.println(
+          String.format(
+              "- No tables are identified for %s. Please check logs for more info.",
+              deleteSourceCatalogTables ? "migration" : "registration"));
+      return;
+    }
+    printWriter.println(
+        String.format(
+            "- Identified %d tables for %s by dry-run. These identifiers are also written into %s. "
+                + "You can use this file with `--identifiers-from-file` option.",
+            result.registeredTableIdentifiers().size(),
+            deleteSourceCatalogTables ? "migration" : "registration",
+            DRY_RUN_FILE));
+
+    printWriter.println(String.format("%nDetails: "));
+    printWriter.println(
+        String.format(
+            "- Identified these tables for %s by dry-run:",
+            deleteSourceCatalogTables ? "migration" : "registration"));
+    printWriter.println(result.registeredTableIdentifiers());
+  }
+
+  private static void writeToFile(Path filePath, List<TableIdentifier> identifiers) {
+    List<String> identifiersString =
+        identifiers.stream().map(TableIdentifier::toString).collect(Collectors.toList());
+    try {
+      Files.write(filePath, identifiersString);
+    } catch (IOException e) {
+      throw new RuntimeException("Failed to write the file:" + filePath, e);
+    }
+  }
+
+  private Path pathWithOutputDir(String fileName) {
+    if (outputDirPath == null) {
```

Comment:
`outputDirPath` should never be null, eliminating the need for this whole function